test/frontend_test.dart

```diff
@@ -0,0 +1,457 @@
+                  'dropdown-item/span[contains(text(), "45 perc")]'));
+          await thirtyMinFilterOption.click();
+          await waitForAngular(driver);
+          await sleep2();
+          var filteredVideoItemsLength = await driver
+              .findElements(const By.cssSelector(
+                  'video-summary-list > div.training-type-list > div '
+                  '> video-summary > div.video-summary'))
+              .length;
+          expect(await isXAtLeastValueOfY(filteredVideoItemsLength, 6), true);
+        });
+
+        test('text filter options work', () async {
+          var textFilter = await driver.findElement(
+              const By.xpath("*//input[contains(@class, 'search-input')]"));
+          await textFilter.click();
+          await textFilter.sendKeys("FORMATORNA");
+          await waitForAngular(driver);
+          await sleep2();
+          var filteredVideoItemsLength = await driver
+              .findElements(const By.cssSelector(
+                  'video-summary-list > div.training-type-list > div '
+                  '> video-summary > div.video-summary'))
+              .length;
+          expect(await isXAtLeastValueOfY(filteredVideoItemsLength, 2), true);
+        });
+      });
+    });
+
+    group("product page megvasarlom tests |", () {
+      setUpAll(() async {
+        await loginUserWithCredentials(
+            newUserEmail, allUserPassword, driver, true);
+        await waitForAngular(driver);
+        await navigateToPage(driver, 'profile/product/probaidoszak');
+        await waitForAngular(driver);
+        await sleep5();
+      });
+
+      test('clicking on megvasarlom throws no errors | the gwl', () async {
+        bool hasErrors = false;
+        await clickMegvasarlom(
+            driver, 'product/view/a-nagy-sulyrakezdo-program-kezdo-szint');
+        hasErrors = await checkForConsoleErrors(driver, 'SEVERE');
+        expect(await hasErrors, false);
+      });
+
+      test('clicking on megvasarlom throws no errors | superhuman program',
+          () async {
+        bool hasErrors = false;
+        await clickMegvasarlom(driver, 'product/view/szuperhas-program');
+        hasErrors = await checkForConsoleErrors(driver, 'SEVERE');
+        expect(await hasErrors, false);
+      });
+
+      test('clicking on megvasarlom throws no errors | dancing mom',
+            () async {
+          bool hasErrors = false;
+          await clickMegvasarlom(driver, 'product/view/tancolj-anyukam-program');
+          hasErrors = await checkForConsoleErrors(driver, 'SEVERE');
+          expect(await hasErrors, false);
+      });
+
+      test('clicking on megvasarlom throws no errors | joga mania',
+            () async {
+          bool hasErrors = false;
+          await clickMegvasarlom(driver, 'product/view/jogamania-program');
+          hasErrors = await checkForConsoleErrors(driver, 'SEVERE');
+          expect(await hasErrors, false);
+      });
+    });
```

Comment:
The tests in this group only check if there are any JS errors after clicking on the purchase button. These should be extended to actually test the purchase workflow, right up to the redirection to the online payment page.